Compatibility with jQuery 1.4.

Index: roundcubemail/program/include/rcube_json_output.php
===================================================================
--- roundcubemail/program/include/rcube_json_output.php (revision 3175)
+++ roundcubemail/program/include/rcube_json_output.php (revision 3202)
@@ -219,5 +219,4 @@
             send_nocacheing_headers();
             header('Content-Type: text/plain; charset=' . $this->get_charset());
-            print '/** ajax response ['.date('d/M/Y h:i:s O')."] **/\n";
         }
 
