--- roundcubemail-0.1~rc1~dfsg.orig/index.php~	2007-05-18 15:11:22.000000000 +0200
+++ roundcubemail-0.1~rc1~dfsg.orig/index.php	2007-05-20 14:12:35.000000000 +0200
@@ -47,7 +47,7 @@
 
 // define global vars
 $OUTPUT_TYPE = 'html';
-$INSTALL_PATH = dirname(__FILE__);
+$INSTALL_PATH = '/var/lib/roundcube';
 $MAIN_TASKS = array('mail','settings','addressbook','logout');
 
 if (empty($INSTALL_PATH))
