Switch to UTF-8 as default charset

Index: b/config/main.inc.php.dist
===================================================================
--- a/config/main.inc.php.dist	2010-07-17 17:33:24.000000000 +0200
+++ b/config/main.inc.php.dist	2010-07-17 17:33:25.000000000 +0200
@@ -215,7 +215,7 @@
 $rcmail_config['mdn_requests'] = 0;
 
 // Use this charset as fallback for message decoding
-$rcmail_config['default_charset'] = 'ISO-8859-1';
+$rcmail_config['default_charset'] = 'UTF-8';
 
 // Make use of the built-in spell checker. It is based on GoogieSpell.
 // Since Google only accepts connections over https your PHP installatation
