Switch to UTF-8 as default charset

Index: roundcube/config/defaults.inc.php
===================================================================
--- roundcube.orig/config/defaults.inc.php	2015-03-16 23:01:16.344835937 +0100
+++ roundcube/config/defaults.inc.php	2015-03-16 23:02:57.229185128 +0100
@@ -861,7 +861,7 @@
 // ----------------------------------
 
 // Use this charset as fallback for message decoding
-$config['default_charset'] = 'ISO-8859-1';
+$config['default_charset'] = 'UTF-8';
 
 // skin name: folder from skins/
 $config['skin'] = 'larry';
