Switch to UTF-8 as default charset

Index: roundcube/config/defaults.inc.php
===================================================================
--- roundcube.orig/config/defaults.inc.php	2015-03-22 20:09:35.330672372 +0100
+++ roundcube/config/defaults.inc.php	2015-03-22 20:09:35.324005639 +0100
@@ -887,7 +887,7 @@
 // ----------------------------------
 
 // Use this charset as fallback for message decoding
-$config['default_charset'] = 'ISO-8859-1';
+$config['default_charset'] = 'UTF-8';
 
 // skin name: folder from skins/
 $config['skin'] = 'larry';
