Switch to UTF-8 as default charset

Index: b/config/main.inc.php.dist
===================================================================
--- a/config/main.inc.php.dist	2010-07-17 17:33:24.000000000 +0200
+++ b/config/main.inc.php.dist	2010-07-17 17:33:25.000000000 +0200
@@ -489,7 +489,7 @@
 // ----------------------------------
 
 // Use this charset as fallback for message decoding
-$rcmail_config['default_charset'] = 'ISO-8859-1';
+$rcmail_config['default_charset'] = 'UTF-8';
 
 // skin name: folder from skins/
 $rcmail_config['skin'] = 'default';
