Switch to UTF-8 as default charset

Index: roundcube/config/defaults.inc.php
===================================================================
--- roundcube.orig/config/defaults.inc.php	2016-05-05 11:00:55.765648355 +0200
+++ roundcube/config/defaults.inc.php	2016-05-05 11:00:55.761648295 +0200
@@ -1004,7 +1004,7 @@
 // ----------------------------------
 
 // Use this charset as fallback for message decoding
-$config['default_charset'] = 'ISO-8859-1';
+$config['default_charset'] = 'UTF-8';
 
 // skin name: folder from skins/
 $config['skin'] = 'larry';
