--- roundcube-0.1/program/js/editor.js~	2008-02-13 04:31:05.000000000 +0100
+++ roundcube-0.1/program/js/editor.js	2008-03-07 09:53:39.000000000 +0100
@@ -21,7 +21,7 @@
                  accessibility_focus : false,
                  apply_source_formatting : true,
                  theme : 'advanced',
-                 plugins : 'emotions,media,nonbreaking,table,searchreplace,spellchecker,visualchars',
+                 plugins : 'emotions,media,nonbreaking,table,searchreplace,visualchars',
                  theme_advanced_buttons1 : 'bold,italic,underline,separator,justifyleft,justifycenter,justifyright,justifyfull,separator,bullist,numlist,outdent,indent,separator,emotions,charmap,code,forecolor,backcolor,fontselect,fontsizeselect, separator,undo,redo,image,media',
                  theme_advanced_buttons2 : '',
                  theme_advanced_buttons3 : '',
