--- roundcube/program/steps/mail/sendmail.inc	2008-03-07 09:52:40.000000000 +0100
+++ roundcube/program/steps/mail/sendmail.inc.new	2008-03-07 11:04:33.000000000 +0100
@@ -351,7 +351,11 @@
   
   // append message to sent box
   if ($store_folder)
-    $saved = $IMAP->save_message($CONFIG[$store_target], $MAIL_MIME->getMessage());
+    {
+    $header_str = $MAIL_MIME->txtHeaders($headers);
+    $msg_body = $header_str."\r\n".$MAIL_MIME->get($message_param);
+    $saved = $IMAP->save_message($CONFIG[$store_target], $msg_body);
+    }
 
   // raise error if saving failed
   if (!$saved)
