--- a/program/include/rcmail.php~	2008-06-07 21:33:07.000000000 +0200
+++ a/program/include/rcmail.php	2008-06-22 13:36:57.000000000 +0200
@@ -474,7 +474,7 @@
   public function autoselect_host()
   {
     $default_host = $this->config->get('default_host');
-    $host = !empty($default_host) ? get_input_value('_host', RCUBE_INPUT_POST) : $default_host;
+    $host = isset($_POST['_host']) ? get_input_value('_host', RCUBE_INPUT_POST) : $default_host;
     
     if (is_array($host)) {
       list($user, $domain) = explode('@', get_input_value('_user', RCUBE_INPUT_POST));
