--- roundcubemail/program/steps/mail/sendmail.inc (revision 1232)
+++ roundcubemail/program/steps/mail/sendmail.inc (revision 1261)
@@ -211,7 +211,4 @@
 
 // additional headers
-$headers['Message-ID'] = $message_id;
-$headers['X-Sender'] = $from;
-
 if ($CONFIG['http_received_header'])
 {
@@ -224,4 +221,7 @@
 }
 
+$headers['Message-ID'] = $message_id;
+$headers['X-Sender'] = $from;
+
 if (!empty($CONFIG['useragent']))
   $headers['User-Agent'] = $CONFIG['useragent'];
