--- roundcubemail-0.1beta2/program/steps/mail/sendmail.inc~	2006-08-01 00:51:23.000000000 +0200
+++ roundcubemail-0.1beta2/program/steps/mail/sendmail.inc	2007-04-08 14:58:54.000000000 +0200
@@ -107,7 +107,7 @@
   $identity_arr['string'] = $from;
 
 // compose headers array
-$headers = array('Date' => date('D, j M Y G:i:s O'),
+$headers = array('Date' => date('D, j M Y H:i:s O'),
                  'From' => $identity_arr['string'],
                  'To'   => rcube_charset_convert($mailto, $input_charset, $message_charset));
 
