Various SQL fixes for 0.7.
--- roundcubemail/SQL/mysql.update.sql	(revision 5468)
+++ roundcubemail/SQL/mysql.update.sql	(revision 5618)
@@ -147,4 +147,6 @@
 
 -- Updates from version 0.6
+
+/*!40014 SET FOREIGN_KEY_CHECKS=0 */;
 
 ALTER TABLE `users` CHANGE `alias` `alias` varchar(128) BINARY NOT NULL;
@@ -210,4 +212,6 @@
 ) /*!40000 ENGINE=INNODB */ /*!40101 CHARACTER SET utf8 COLLATE utf8_general_ci */;
 
+/*!40014 SET FOREIGN_KEY_CHECKS=1 */;
+
 -- Updates from version 0.7-beta
 
--- roundcubemail/SQL/sqlite.update.sql	(revision 5468)
+++ roundcubemail/SQL/sqlite.update.sql	(revision 5615)
@@ -247,5 +247,5 @@
 );
 
-CREATE UNIQUE INDEX ix_searches_user_type_name (user_id, type, name);
+CREATE UNIQUE INDEX ix_searches_user_type_name ON searches (user_id, type, name);
 
 DROP TABLE messages;
--- roundcubemail/SQL/sqlite.initial.sql	(revision 5460)
+++ roundcubemail/SQL/sqlite.initial.sql	(revision 5615)
@@ -148,5 +148,5 @@
 );
 
-CREATE UNIQUE INDEX ix_searches_user_type_name (user_id, type, name);
+CREATE UNIQUE INDEX ix_searches_user_type_name ON searches (user_id, type, name);
 
 -- --------------------------------------------------------
--- roundcubemail/SQL/postgres.update.sql	(revision 5468)
+++ roundcubemail/SQL/postgres.update.sql	(revision 5627)
@@ -128,5 +128,5 @@
 );
 
-DROP SEQUENCE messages_ids;
+DROP SEQUENCE message_ids;
 DROP TABLE messages;
 
