--- roundcubemail-0.1-dep/config/db.inc.php.dist~	2008-03-03 22:32:15.000000000 +0100
+++ roundcubemail-0.1-dep/config/db.inc.php.dist	2008-03-05 21:07:28.000000000 +0100
@@ -27,7 +27,7 @@
 $rcmail_config['db_dsnr'] = '';
 
 // database backend to use (only db or mdb2 are supported)
-$rcmail_config['db_backend'] = 'mdb2';
+$rcmail_config['db_backend'] = 'db';
 
 // maximum length of a query in bytes
 $rcmail_config['db_max_length'] = 512000;  // 500K
