Use packaged libjs-jquery-ui.

Index: roundcube/plugins/jqueryui/jqueryui.php
===================================================================
--- roundcube.orig/plugins/jqueryui/jqueryui.php	2012-09-29 11:30:56.195077966 +0200
+++ roundcube/plugins/jqueryui/jqueryui.php	2012-09-29 11:31:49.528144533 +0200
@@ -21,7 +21,7 @@
     $this->load_config();
 
     // include UI scripts
-    $this->include_script("js/jquery-ui-$version.custom.min.js");
+    $this->include_script("js/jquery-ui.min.js");
 
     // include UI stylesheet
     $skin = $rcmail->config->get('skin', 'default');
