Index: roundcube-0.1~rc2/program/steps/mail/sendmail.inc
===================================================================
--- roundcube-0.1~rc2.orig/program/steps/mail/sendmail.inc	2007-11-26 11:42:36.000000000 +0100
+++ roundcube-0.1~rc2/program/steps/mail/sendmail.inc	2007-11-26 11:44:04.000000000 +0100
@@ -108,7 +108,7 @@
     if (! in_array($image_name, $included_images))
       {
       // add the image to the MIME message
-      $img_file = INSTALL_PATH . '/' . $searchstr . $image_name;
+      $img_file = '/usr/share/tinymce/www/plugins/emotions/images/' . $image_name;
       if(! $mime_message->addHTMLImage($img_file, 'image/gif', '', true, '_' . $image_name))
         $OUTPUT->show_message("emoticonerror", 'error');
 
